lib/workers/repository/update/branch/schedule.ts

```diff
@@ -81,29 +87,25 @@ export function hasValidSchedule(
 }
 
 function cronMatches(cron: string, now: DateTime): boolean {
+  // it will always parse because it is checked beforehand
   const parsedCron = parseCron(cron);
 
-  // istanbul ignore if: doesn't return undefined but type will include undefined
-  if (!parsedCron) {
-    return false;
-  }
-
-  if (parsedCron.hours.indexOf(now.hour) === -1) {
+  if ((parsedCron.fields.hour as number[]).indexOf(now.hour) === -1) {
     // Hours mismatch
     return false;
   }
 
-  if (parsedCron.days.indexOf(now.day) === -1) {
+  if ((parsedCron.fields.dayOfMonth as number[]).indexOf(now.day) === -1) {
```

Comment:
Why the cast?

Reply:
Type of dayOfMonth is `<ReadonlyArray>|1|2|3|4|5|6|...upto 31` whereas now.hour is number

Reply:
I would prefer if we avoid `as` casts as they might change that type later and we wouldn't know